.github/workflows/ci.yml

```diff
@@ -369,34 +369,24 @@ jobs:
     name: coverage ${{ matrix.os }}
     strategy:
       matrix:
-        os: ["windows-latest", "macos-14", "ubuntu-latest"]  # first available arm macos runner
+        # macos-14 is first available arm macos runner
+        # it's the fastest job so use it on PR instead of ubuntu to get faster feedback
+        os: ${{ (github.event_name == 'pull_request' && !contains(github.event.pull_request.labels.*.name, 'CI-build-full')) && fromJson('["macos-14"]') || fromJson('["windows-latest", "macos-14", "ubuntu-latest"]') }}
     runs-on: ${{ matrix.os }}
     steps:
-      - if: ${{ github.event_name == 'pull_request' && matrix.os != 'ubuntu-latest' }}
-        id: should-skip
-        shell: bash
-        run: echo 'skip=true' >> $GITHUB_OUTPUT
       - uses: actions/checkout@v4
-        if: steps.should-skip.outputs.skip != 'true'
```

Comment:
I vaguely remember trying to remove these and failing for some reason. Do you remember why we had these in the first place?

Reply:
We can't reference the `matrix` in the usual `job.if` clause, so maybe we just hadn't worked out that we could use `fromJson()` to manage the matrix conditionally.